src/asymmetric_cipher.c

```diff
@@ -24,6 +27,9 @@ struct p11prov_rsaenc_ctx {
 
     CK_MECHANISM_TYPE mechtype;
     CK_RSA_PKCS_OAEP_PARAMS oaep_params;
+    int ossl_id;            /* real OSSL padding mode */
+    int client_version;     /* RSA_PKCS1_WITH_TLS_PADDING: client_version */
+    int alt_version;        /* RSA_PKCS1_WITH_TLS_PADDING: negociated_version */
```

Comment:
These are passed as int by openssl for convenience, but are actually each a pair of 1 byte values.
I would either declare them as uint16_t or even better already split thme into the component bytes and declare a pair of uint8_t for each.